contributing/topics/maintainer-changelog.md

```diff
@@ -52,3 +52,46 @@ BUG FIXES:
 * `azurerm_linux_function_app` - correctly deduplicate user `app_settings` [GH-12345]
 * `azurerm_windows_function_app_slot` - correctly deduplicate user `app_settings` [GH-12345]
 ```
+
+## Automated Changelog Guide
+For maintainers, when reviewing and merging a PR that warrants a changelog entry, the changelog automation flow is documented below.
+
+In the Extended description box of the merge commit message type the changelog entry. 
+
+Example: ```[BUG] * Data Source: `azurerm_data_source` - prevent a possible crash by setting `queue_name` correctly```
+
+The Github PR number (like `[GH-12345]`) will be appended by the automation.
+
+The options for the automation are:
+
+* `[BUG]`
+
+* `[ENHANCEMENT]`
+
+* `[FEATURE]`
+
+> **Note:** Breaking changes need to be added manually to the open changelog PR by editing the branch the changelog PR is open on. 
+
+
+After pressing “Confirm squash and merge”, the automation will kick off. 
+
+1. It will pull the merge commit message and append the PR number `[GH-{number}]` 
+
+2. It will check for the keywords `[BUG]`, `[ENHANCEMENT]`, `[FEATURE]`
+3. If a keyword is used, a changelog entry will be made
```

Comment:
```suggestion
2. It will check for the keywords `[BUG]`, `[ENHANCEMENT]`, `[FEATURE]`

3. If a keyword is used, a changelog entry will be made
```